src/cpp/server/server_cc.cc

```diff
@@ -152,19 +152,24 @@ class Server::SyncRequest final : public internal::CompletionQueueTag {
     GPR_ASSERT(cq_ && !in_flight_);
     in_flight_ = true;
     if (tag_) {
-      GPR_ASSERT(GRPC_CALL_OK ==
-                 grpc_server_request_registered_call(
-                     server, tag_, &call_, &deadline_, &request_metadata_,
-                     has_request_payload_ ? &request_payload_ : nullptr, cq_,
-                     notify_cq, this));
+      if (grpc_server_request_registered_call(
```

Comment:
Minor nit: To make it easier to read, could you change this to  `if (GRPC_CALL_OK != grpc_server_request_registered_call(...)) {`

When I was reading it initially, i was really confused as to why you were 'TearingDown' the request if request call was successful